src/api/context/CardContext.tsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useMemo } from 'react';
+import { Product } from '../../types/Product';
+import { useLocalStorage } from '../../helpers/useLocalStorage';
+
+interface State {
+  cardProducts: Product[],
+  setCardProducts: (products: Product[]) => void,
+  handleAddToCard: (product: Product) => void,
+  visbleProducts: Product[],
+  removeProduct: (productId: string) => void,
+  countProducts: (productId: string) => number,
+  increment: (product: Product) => void,
+  decrement: (productId: string) => void,
+}
+
+export const CardContext = React.createContext<State>({
+  cardProducts: [],
+  setCardProducts: () => {},
+  handleAddToCard: () => {},
+  visbleProducts: [],
+  removeProduct: () => {},
+  countProducts: () => 0,
+  increment: () => {},
+  decrement: () => {},
+});
+
+interface Props {
+  children: React.ReactNode,
+}
+
+export const CardProvider: React.FC<Props> = ({ children }) => {
+  const [
+    cardProducts,
+    setCardProducts,
+  ] = useLocalStorage<Product[]>('card', []);
+
+  const isInCard: string[] = [];
+
+  const visbleProducts = cardProducts.filter(item => {
+    if (isInCard.includes(item.id)) {
+      return false;
+    }
+
+    isInCard.push(item.id);
+
+    return true;
+  });
+
+  const countProducts = (productId: string) => {
+    return cardProducts.filter(prod => prod.id === productId).length;
+  };
+
+  const handleAddToCard = (product: Product) => {
```

Comment:
you may use `increment`, `removeProduct` (maybe rename something if needed), here to avoid code duplication